contracts/Multisig.vy

```diff
@@ -0,0 +1,153 @@
+transactions: DynArray[Transaction, 10000]
+
+#######################################################################################  
+# INITIALIZATION FUNCTION
+#######################################################################################  
+@external
+def __init__ (_numConfirmationsRequired: uint256, _owners: DynArray[address,100]):
+    assert len(_owners) > 0, "Error: No Owners"
+    assert _numConfirmationsRequired > 0 and _numConfirmationsRequired <= len(_owners)
+    self.numConfirmationsRequired = _numConfirmationsRequired
+    for o in _owners:
+        assert o != empty(address)
+        assert self.isOwner[o] == False, "Owner Not Unique"
+        self.isOwner[o] = True
+        self.owners.append(o)
+
+#######################################################################################  
+# EXTERNAL FUNCTIONS
+####################################################################################### 
+@external
+@payable
+def __default__():
+    log Deposit(msg.sender, msg.value, self.balance)
+
+@external
+def submitTransaction( _to: address, _val: uint256, _data: Bytes[256]):
+    assert self.isOwner[msg.sender] == True, "Not an Owner"
+    txIndex: uint256 = len(self.transactions)
+    self.transactions.append(Transaction({to:_to, val:_val, data: _data, executed: False, numConfirmations: 0}))    
+
+@external 
+def confirmTransaction(txIndex: uint256):
+    assert self.isOwner[msg.sender]==True, "Not an Owner"
+    assert txIndex < len(self.transactions), "Does Not Exist"
+    assert self.transactions[txIndex].executed == False, "Transaction Already Executed"
+    assert self.isConfirmed[txIndex][msg.sender] == False, "Transaction Already Confirmed"
+    # increment the number of tx confirmations
+    self.transactions[txIndex].numConfirmations += 1
+    # take note of the msg sender signing the tx
+    self.isConfirmed[txIndex][msg.sender] = True
+    log ConfirmTransaction(msg.sender, txIndex)
+
+
+@payable
+@external
+def executeTransaction(txIndex: uint256) -> Bytes[256]:
+    assert self.isOwner[msg.sender]==True, "Not an Owner"
+    assert txIndex < len(self.transactions), "Does Not Exist"
+    assert self.transactions[txIndex].executed == False, "Transaction Already Executed"
+    assert self.transactions[txIndex].numConfirmations >= self.numConfirmationsRequired, "Insufficent Confirmations"
+    assert self.transactions[txIndex].val <= self.balance, "Insufficient multisig balance"
+    success: bool = False
+    response: Bytes[256] = b""
+    success, response = raw_call(
+        self.transactions[txIndex].to, 
+        self.transactions[txIndex].data,
+        max_outsize=256, # what's that?
+        value = self.transactions[txIndex].val,
+        revert_on_failure = False # what is this and why we need it to be False?
+    )
+    assert success
+    # label the tx as executed after success is confirmed
+    self.transactions[txIndex].executed = True

```

Comment:
This is a dangerous anti-pattern! State must **always be updated before making external calls**. https://mirror.xyz/crypdough.eth/hR_kLAqmqZy1xUTyRrOLVPoR-4Zr--79hRc0p98CWhs